libraries/AP_Math/polygon.cpp

```diff
@@ -109,3 +111,84 @@ template bool Polygon_outside<int32_t>(const Vector2l &P, const Vector2l *V, uns
 template bool Polygon_complete<int32_t>(const Vector2l *V, unsigned n);
 template bool Polygon_outside<float>(const Vector2f &P, const Vector2f *V, unsigned n);
 template bool Polygon_complete<float>(const Vector2f *V, unsigned n);
+
+
+/*
+  determine if the polygon of N verticies defined by points V is
+  intersected by a line from point p1 to point p2
+ */
+bool Polygon_intersects(const Vector2f *V, unsigned N, const Vector2f &p1, const Vector2f &p2, Vector2f &intersection)
```

Comment:
This should ideally be templated so we can leverage it against the Vector2l polygon's we have. (Quite useful for path generation for a survey grid for instance)